packages/gasket-react-intl/EXAMPLES.md

```diff
@@ -0,0 +1,521 @@
+      <IntlProvider locale={locale} messages={messages}>
+        {children}
+      </IntlProvider>
+    </div>
+  );
+}
+
+const CustomIntlMessagesProvider = withMessagesProvider(intlManager)(CustomIntlProvider);
+
+export default function App({ children }) {
+  return (
+    <CustomIntlMessagesProvider locale="fr-FR">
+      {children}
+    </CustomIntlMessagesProvider>
+  );
+}
+```
+
+## useMessages
+
+Hook to get all currently loaded messages.
+
+```javascript
+import { useMessages } from '@gasket/react-intl';
+
+export default function MyComponent() {
+  const messages = useMessages();
+
+  return (
+    <div>
+      <p>{messages.welcome}</p>
+      <p>{messages.description}</p>
+      <p>Available messages: {Object.keys(messages).join(', ')}</p>
+    </div>
+  );
+}
+```
+
+### With Message Validation
+
+```javascript
+import { useMessages } from '@gasket/react-intl';
+
+export default function SafeComponent() {
+  const messages = useMessages();
+
+  // Safely access messages with fallbacks
+  const welcomeMessage = messages.welcome || 'Welcome';
+  const hasErrorMessages = 'error' in messages;
+
+  return (
+    <div>
+      <h1>{welcomeMessage}</h1>
+      {hasErrorMessages && (
+        <div className="error">
+          {messages.error}
+        </div>
+      )}
+    </div>
+  );
+}
+```
+
+## useLocaleFile
+
+Hook to load locale files and get their loading status.
+
+### Single Locale File
+
+```javascript
+import { useLocaleFile, LocaleFileStatus } from '@gasket/react-intl';
+
+export default function ProductPage() {
+  const status = useLocaleFile('/locales/products');
+
+  if (status === LocaleFileStatus.loading) {
+    return <div>Loading product translations...</div>;
+  }
+
+  if (status === LocaleFileStatus.error) {
+    return <div>Failed to load translations</div>;
+  }
+
+  if (status === LocaleFileStatus.loaded) {
+    return <div>Product page content...</div>;
+  }
+
+  return null;
+}
+```
+
+### Multiple Locale Files
+
+```javascript
+import { useLocaleFile, LocaleFileStatus } from '@gasket/react-intl';
+
+export default function CheckoutPage() {
+  const status = useLocaleFile('/locales/checkout', '/locales/payment', '/locales/shipping');
+
+  // Returns the "lowest" status (least loaded state)
+  switch (status) {
+    case LocaleFileStatus.loading:
+      return <div>Loading checkout translations...</div>;
+    case LocaleFileStatus.error:
+      return <div>Translation error occurred</div>;
+    case LocaleFileStatus.loaded:
+      return <div>Checkout form...</div>;
+    default:
+      return <div>Initializing...</div>;
+  }
+}
+```
+
+### Conditional Loading
+
+```javascript
+import { useLocaleFile, LocaleFileStatus, useMessages } from '@gasket/react-intl';
+
+export default function ConditionalComponent({ showAdvanced }) {
+  const basicStatus = useLocaleFile('/locales/basic');
+  const advancedStatus = showAdvanced ? useLocaleFile('/locales/advanced') : LocaleFileStatus.loaded;
```

Comment:
I don't think you can conditionally use hooks like this. This conditional loading is a bad example. Just delete it's not really useful. Components can conditionally use other components, which in turn could use different locale files.